### PR TITLE
docs: update Readme

### PR DESCRIPTION
**Pull request by roeap**
# Description

With summit coming up I thought we might update our README, since delta-rs has evolved quite a bit since the README was first written...

Just opening the Draft to get feedback on the general "patterns" i.e. how the tables are formatted, how detailed we want to show the features and mostly the looks of the header.

Also hoping our community experts may have some content they wat to add here 😆.

cc @dennyglee @MrPowers @wjones127 @rtyler @houqp @fvaleye

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+write_deltalake("./data/delta", df)
+
+# load data from delta table
+dt = DeltaTable("./data/delta")
+df2 = dt.to_pandas()
+
+assert df == df2
+```
+
+The same table written cal also be loaded using the core rust crate.
```
Comment (wjones127) [inline README.md:83]:
```suggestion
The same table written can also be loaded using the core Rust crate:
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+| Reader Version | Requirement                         | Status |
+| -------------- | ----------------------------------- | ------ |
+| Version 2      | Collumn Mapping                     |        |
+| Version 3      | Table Features (requires reader V7) |        |
+
+[datafusion]: https://github.com/apache/arrow-datafusion
+[ballista]: https://github.com/apache/arrow-ballista
+[polars]: https://github.com/pola-rs/polars
+[open]: https://cdn.jsdelivr.net/gh/Readme-Workflows/Readme-Icons@main/icons/octicons/IssueOpened.svg
+[done]: https://cdn.jsdelivr.net/gh/Readme-Workflows/Readme-Icons@main/icons/octicons/IssueClosed.svg
```
Comment (wjones127) [inline README.md:165]: Fancy!

File: python/pyproject.toml [hunk 3]
```diff
@@ -10,9 +10,12 @@ license = {file = "LICENSE.txt"}
 requires-python = ">=3.7"
 keywords = ["deltalake", "delta", "datalake", "pandas", "arrow"]
 classifiers = [
-    "Development Status :: 3 - Alpha",
+    "Development Status :: 4 - Beta",
```
Comment (wjones127) [inline python/pyproject.toml:13]: 👍

File: README.md
Comment (wjones127) [inline README.md:1]: What do you think of adding a "Powered By" section and listing Apache Arrow and DataFusion?

File: REA_DME.adoc
Comment (wjones127) [inline REA_DME.adoc:1]: Are we to delete this?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  </a>
+  <a href="https://pypi.python.org/pypi/deltalake">
+    <img alt="Deltalake" src="https://img.shields.io/pypi/v/deltalake.svg?style=flat-square&color=F75101&logo=pypi" >
+  </a>
+  <a href="https://pypi.python.org/pypi/deltalake">
+    <img alt="Deltalake" src="https://img.shields.io/pypi/pyversions/deltalake.svg?style=flat-square&color=00ADD4&logo=python">
+  </a>
+  <a target="_blank" href="https://go.delta.io/slack">
+    <img alt="#delta-rs in the Delta Lake Slack workspace" src="https://img.shields.io/badge/slack-delta-blue.svg?logo=slack&style=flat-square&color=F75101">
+  </a>
```
Comment (wjones127) [inline README.md:37]: Could we add back a "Get involved section"?

Could list:

* Join the slack workspace (the button is nice, but it's kind of small)
* Report an issue
* Looking to contribute? See our list of good first issues (https://github.com/delta-io/delta-rs/contribute)

**Review by wjones127 — COMMENTED** [review]
This is looking better. I had a few suggestions

File: README.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+- [AWS SDK for Pandas](https://github.com/aws/aws-sdk-pandas)
+
+## Features
+
+### Cloud Integrations
+
+| Storage              |         Rust          |        Python         | Comment                             |
+| -------------------- | :-------------------: | :-------------------: | ----------------------------------- |
+| Local                |        ![done]        |        ![done]        |                                     |
+| S3 - AWS             |        ![done]        |        ![done]        | requires lock for concurrent writes |
```
Comment (dennyglee) [inline README.md:119]: Should we point to the docs to explain how to spin up DynamoDB for the lock?

File: README.md (hunk 2 above)
Comment (dennyglee) [reply README.md:165]: Seriously awesome :)

**Review by dennyglee — CHANGES_REQUESTED** [review]
This is really nice, thanks for tackling this.  Minor suggestions noted.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  </a>
+  <a href="https://pypi.python.org/pypi/deltalake">
+    <img alt="Deltalake" src="https://img.shields.io/pypi/pyversions/deltalake.svg?style=flat-square&color=00ADD4&logo=python">
+  </a>
+  <a target="_blank" href="https://go.delta.io/slack">
+    <img alt="#delta-rs in the Delta Lake Slack workspace" src="https://img.shields.io/badge/slack-delta-blue.svg?logo=slack&style=flat-square&color=F75101">
+  </a>
+</p>
+
+The delta-rs project aims to unlock the power of the Deltalake for as many users and projects as possible
```
Comment (dennyglee) [inline README.md:40]: Please update to "Delta Lake"

File: README.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+by providing native low level APIs aimed at developers and integrators, as well as a high level operations
+API that lets you query, inspect, and operate your Deltalake with ease.
```
Comment (dennyglee) [inline README.md:42]: Please update to "Delta Lake"

File: README.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+## Table of contents
+
+- [Quick Start](#quick-start)
+- [Integartions](#integrations)
+- [Features](#features)
+
+## Quick Start
+
+The deltalake library aim to adopt familiar patterns from other libraries in data processing,
```
Comment (dennyglee) [inline README.md:64]: Suggest either "The Delta Lake library aims to..." or "The `deltalake` library aims to..."

File: README.md
Comment (roeap) [reply README.md:1]: should we then also submit a PR to the official arrow "pwered_by" section? 

https://arrow.apache.org/powered_by/

File: REA_DME.adoc
Comment (roeap) [reply REA_DME.adoc:1]: we are :) - just keeping it around for refence, until all content is ported over..

**Comment by MrPowers** [comment]
Looks like some great improvements!

File: README.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+    Ok(())
+}
+```
+
+## Get Involved
+
+We encourage you to reach out, and are [commited](https://github.com/delta-io/delta-rs/blob/main/CODE_OF_CONDUCT.md)
+to provide a welcoming community.
+
+- [Join us in our slack worksapce](https://go.delta.io/slack)
```
Comment (wjones127) [inline README.md:107]:
```suggestion
- [Join us in our Slack workspace](https://go.delta.io/slack)
```

**Review by wjones127 — COMMENTED** [review]
on typo

File: README.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+| Optimize - Z-order    |       ![done]       |       ![done]       | Place similar data into the same file |
+| Merge                 | [![open]][merge-rs] | [![open]][merge-py] |                                       |
+| FS check              |       ![done]       |                     | Remove corrupted files from table     |
+
+### Protocol Support Level
+
+| Writer Version | Requirement                                   |        Status        |
+| -------------- | --------------------------------------------- | :------------------: |
+| Version 2      | Append Only Tables                            |  [![open]][roadmap]  |
+| Version 2      | Column Invatiants                             |       ![done]        |
```
Comment (MrPowers) [inline README.md:162]: Little typo here (Invatiants => Invariants)

Comment (rtyler) [inline README.md:162]:
```suggestion
| Version 2      | Column Invariants                             |       ![done]        |
```

**Review by rtyler — DISMISSED** [review]

**Review by rtyler — APPROVED** [review]

**Pull request merged**